update README + Site - release 0.4

miguelangel-dev opened the pull request:
### Description
Update the `readme` and `microsite` of nef to show the news in release 0.4.

miguelangel-dev committed:
update README with the carthage compatibility + installation sources

miguelangel-dev committed:
add assets for new nef playground

miguelangel-dev committed:
aligned site with the README

miguelangel-dev committed:
add a page in site about how to use nef library

truizlop commented on README.md:
```diff
@@ -5,42 +5,65 @@
 </p>
 
-`nef`, short for [Nefertiti](https://en.wikipedia.org/wiki/Nefertiti), mother of Ankhesenamun, is a command line tool to ease the creation of documentation in the form of Xcode Playgrounds. It provides compile-time verification of documentation, exports it in Markdown format that can be consumed by [Jekyll](https://jekyllrb.com/) to generate websites, and export [Carbon](https://carbon.now.sh/) s ...
+`nef`, short for [Nefertiti](https://en.wikipedia.org/wiki/Nefertiti), mother of Ankhesenamun, is a set tool to ease the **creation of documentation** in the form of Xcode Playgrounds. It provides compile-time **verification** of documentation, exports it in **Markdown** format that can be consumed by [Jekyll](https://jekyllrb.com/) to generate websites, and export [Carbon](https://carbon.now.sh/ ...
```
```suggestion
`nef`, short for [Nefertiti](https://en.wikipedia.org/wiki/Nefertiti), mother of Ankhesenamun, is a toolset to ease the **creation of documentation** in the form of Xcode Playgrounds. It provides compile-time **verification** of documentation, exports it in **Markdown** format that can be consumed by [Jekyll](https://jekyllrb.com/) to generate websites, and export [Carbon](https://carbon.now.sh/) snippets for a given Xcode Playground.
```

truizlop commented on README.md:
```diff
@@ -5,42 +5,65 @@
+
+## 💻 Installation
+
+📟 **Using** [**Homebrew**](https://github.com/bow-swift/homebrew-nef) (preferred)
```
Why not using `###` syntax to mark sub sections under `Installation`?

miguelangel-dev replied:
Not sure why I do not use it; changed! Thanks

truizlop commented on README.md:
````diff
@@ -51,18 +74,28 @@ By default, `nef` can create an Xcode Playground with support for [Bow](http://b
+➜ nef playground --platform osx
+```
 
 ###
````
What are these empty section titles?

miguelangel-dev replied:
these are new lines, to add v-spacing between lines. ¯\_(ツ)_/¯

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+// nef:begin:hidden
+import Bow
+Nef.Playground.needsIndefiniteExecution(false)
+// nef:end
```
Not needed in this page.

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+/*:
+ ## How to use `nef` library?
+ 
+ As you know, from Xcode 11, you can integrate package dependencies using SPM to share code between projects or even use third-party libraries. You can read more about it in Apple's article [Adding package dependencies to your app](https://developer.apple.com/documentation/xcode/adding_package_dependencies_to_your_app)
```
Use full name on first appearance at least, with acronym for further reference.

```suggestion
 As you know, from Xcode 11, you can integrate package dependencies using Swift Package Manager (SPM) to share code between projects or even use third-party libraries. You can read more about it in Apple's article [Adding package dependencies to your app](https://developer.apple.com/documentation/xcode/adding_package_dependencies_to_your_app)
```

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+ 
+ As you know, from Xcode 11, you can integrate package dependencies using SPM to share code between projects or even use third-party libraries. You can read more about it in Apple's article [Adding package dependencies to your app](https://developer.apple.com/documentation/xcode/adding_package_dependencies_to_your_app)
+
+ In `nef` [version 0.4](https://github.com/bow-swift/nef/releases/tag/0.4.0), we have modularized the core so that you can use nef as in CLI as in your macOS projects. Taking advantage of the integration of SPM in Xcode, you can easily import `nef` in your project.
```
```suggestion
 Starting on `nef` [version 0.4](https://github.com/bow-swift/nef/releases/tag/0.4.0), we have modularized the core so that you can use nef as in library as in your macOS projects. Taking advantage of the integration of SPM in Xcode, you can easily import `nef` in your project.
```

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+
+ In `nef` [version 0.4](https://github.com/bow-swift/nef/releases/tag/0.4.0), we have modularized the core so that you can use nef as in CLI as in your macOS projects. Taking advantage of the integration of SPM in Xcode, you can easily import `nef` in your project.
+ 
+ Just choose like package repository `https://github.com/bow-swift/nef.git` and Xcode will do the rest.
```
```suggestion
 Just choose as package repository `https://github.com/bow-swift/nef.git` and Xcode will do the rest.
```

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+ 
+ ![](/assets/nef-xcode-library.png)
+ 
+ Once Xcode solves the dependencies, you can import wherever you want, as a native framework:
```
```suggestion
 Once Xcode resolves the dependencies, you can import wherever you want, as a native framework:
```

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+ 
+ ### Jekyll
+ 
+ When you need to render a content that combine Markdown, code, and [**nef commands**](https://github.com/bow-swift/nef#-generating-markdown-files-for-jekyll) with the Jekyll style.
```
```suggestion
 When you need to render content that combines Markdown, code, and [**nef commands**](https://github.com/bow-swift/nef#-generating-markdown-files-for-jekyll) with the Jekyll style.
```

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
````diff
@@ -0,0 +1,111 @@
+ ```
+ 
+ 
+ You can find anothers helpers for drawing `Views` with a Carbon style.
````
```suggestion
 You can find other helpers for drawing `Views` with a Carbon style.
```

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+/*:
+ ## Usage
+ 
+ In this section, we will see how we can use the nef's API. As an example, we will take advantage of the `Carbon API`.
```
```suggestion
 In this section, we will see how we can use the nef API. As an example, we will take advantage of the `Carbon API`.
```

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
````diff
@@ -0,0 +1,111 @@
+ In this section, we will see how we can use the nef's API. As an example, we will take advantage of the `Carbon API`.
+
+ Firstly, we need to create a model with Carbon configuration and the style.
+ ```swift
+ let model = CarbonModel(code: """
+                               import nef
+                               let library = 'nef library is super cool!'
+                               """,
+                         style: CarbonStyle(background: .bow,
+                                            theme: .dracula,
+                                            size: .x1,
+                                            fontType: .firaCode,
+                                            lineNumbers: true, watermark: true))
+ ```
````
This code could change in the future and we won't notice it. We can rely on local Cocoapods to import `nef` as a library on this playground and have this code compile-checked.

miguelangel-dev replied:
Yep, I thought about it, but as we do not have right now the library available, I postponed it for the next release, together with an update to new playground system.

truizlop commented on contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift:
```diff
@@ -0,0 +1,111 @@
+ 
+ In the output path, you can find the result:
+ 
+  ![](/assets/nef-library-example.png)
```
Write an alt text for better accessibility.

miguelangel-dev replied:
done

miguelangel-dev committed:
Update README.md

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

miguelangel-dev committed:
Comments in PR

Update contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>

Update contents/Documentation/Tutorials.playground/Pages/How to use nef library?.xcplaygroundpage/Contents.swift

Co-Authored-By: Tomás Ruiz-López <truizlop@gmail.com>
